superset/assets/spec/javascripts/sqllab/VisualizeModal_spec.jsx

```diff
@@ -66,47 +77,230 @@ describe('VisualizeModal', () => {
     expect(wrapper.find(Modal)).to.have.length(1);
   });
 
-  describe('visualize', () => {
+  describe('setStateFromProps', () => {
     const wrapper = getVisualizeModalWrapper();
+    const sampleQuery = queries[0];
 
-    wrapper.setState({
-      chartType: mockChartTypeBarChart,
-      columns: mockColumns,
-      datasourceName: 'mockDatasourceName',
+    it('should have valid query prop', () => {
```

Comment:
setStateFromProps required valid props as parameter, otherwise it should not change state.
So I add a test that pass invalid paramters, and expects no state change.

The old test description is mis-leading. I will give it a meaningful description.